### PR TITLE
Add ESLint

### LINKED ISSUE #763: Request for JSCS
Hi @rhutchison,

Can you please create a PR for the adding the JSCS (https://github.com/jscs-dev/node-jscs) support? I'd like to push this in along with JSHint that we have today to enforce the code style standards.

I know that you included part of that in another PR that we asked to take out, so there's no point for me to re-do any of the work you already have pending there.
Thanks!

### PR DESCRIPTION
**Pull request by ilanbiala**
Fixes #763.

@codydaig @mleanos @lirantal made a new PR so I didn't have to work off master on my fork.

### CODE REVIEW

**Comment by ilanbiala** [comment]
@mleanos fixed Gulp test, it was actually an issue with me leaving out `.eslint`.

File: gruntfile.js [hunk 1]
```diff
@@ -101,6 +101,10 @@ module.exports = function (grunt) {
         }
       }
     },
+    eslint: {
+      options: {},
```
Comment (lirantal) [inline gruntfile.js]: if there are no options that we're passing can we just not include it?

**Comment by lirantal** [comment]
@ilanbiala what formatting does this eslint config compatible with? is that for the current project formatting until we decide on a final one?

File: gruntfile.js (hunk 1 above)
Comment (ilanbiala) [inline gruntfile.js]: @lirantal we may want to add options once we finalize formatting, so I figured I would just leave it.

**Comment by ilanbiala** [comment]
@lirantal it's pretty close to the formatting we have right now. Obviously we can add any other specific stuff once we discuss it in the code formatting issue. This PR is meant more for just adding in support and we'll tweak all the rules later in a discussion.

**Comment by ilanbiala** [comment]
@lirantal let me know if I can merge, I'll hold off just to make sure.

**Comment by codydaig** [comment]
LGTM!

**Comment by mleanos** [comment]
Tested, and LGTM.

I'm not completely clear on these eslint formatting settings, but based off of the fact that the only formatting changes that were required in the other PR were minor spacing issues, then I think it is line with our current coding styles. 

Once we decide on any changes to our styles, we can adjust these settings. But for now, this is probably good to merge to get our initial ESLint support out.

File: gruntfile.js (hunk 1 above)
Comment (lirantal) [inline gruntfile.js]: If it's redundant we can remove it until we add something there.

**Comment by lirantal** [comment]
@ilanbiala Let's go for it. It's better to have something, than nothing at all or wait for the final coding convention to finalize, which god only knows when will it happen.

**Pull request merged**